### PR TITLE
fix: Add MIRACL reranking

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
These is just a few minor changes to #641 

Still need to add the results from running the model

### CODE REVIEW

File: mteb/tasks/Reranking/multilingual/MIRACLReranking.py [hunk 1]
```diff
@@ -1,38 +1,277 @@
...
...    abstract = "{MIRACL is a multilingual dataset for ad hoc retrieval across 18 languages that collectively encompass over three billion native speakers around the world. This resource is designed to support monolingual retrieval tasks, where the queries and the corpora are in the same language. In total, we have gathered over 726k high-quality relevance judgments for 78k queries over Wikipedia in these languages, where all annotations have been performed by native speakers hired by our team. MIRACL covers languages that are both typologically close as well as distant from 10 language families and 13 sub-families, associated with varying amounts of publicly available resources. Extensive automatic heuristic verification and manual assessments were performed during the annotation process to control data quality. In total, MIRACL represents an investment of around five person-years of human annotator effort. Our goal is to spur research on improving retrieval across a continuum of languages, thus enhancing information access capabilities for diverse populations around the world, particularly those that have traditionally been underserved. MIRACL is available at http://miracl.ai/.}",
```
Comment (dokato) [inline mteb/tasks/Reranking/multilingual/MIRACLReranking.py:53]: Nitpicking, but probably we don't need a full abstract here?

File: mteb/tasks/Reranking/multilingual/MIRACLReranking.py [hunk 2]
```diff
@@ -1,38 +1,277 @@
...
+            fake_qid = str(query_idx)
+            results[fake_qid] = self.rerank(query_emb, docs_emb)
+            qrels[fake_qid] = {
+                str(i): 1 if doc in positive else 0 for i, doc in enumerate(docs)
+            }
+
+        ndcg, _map, recall, precision = RetrievalEvaluator.evaluate(
+            qrels=qrels, results=results, k_values=self.k_values
+        )
+        return {**ndcg, **_map, **recall, **precision}
```
Comment (crystina-z) [inline mteb/tasks/Reranking/multilingual/MIRACLReranking.py:234]: I just saw this. Thanks for taking care of the remaining issues! apologized that I was distracted by other work.

Here it needs to append the "(MIRACL)" after metrics as well?

**Comment by KennethEnevoldsen** [comment]
@crystina-z seems like the dataset is gated. Is this intended?

**Comment by crystina-z** [comment]
heyy @KennethEnevoldsen the access request is removed now. I'm taking a look at where the duplicates come from

**Comment by KennethEnevoldsen** [comment]
I added results - I believe this concludes this PR - I additionally also fixed some broken dataset reference (unrelated, due to failing tests)

**Pull request merged**